### PR TITLE
Backlog 16051 Add structured view for pages/contents

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-16051

## Description

Add structured view for pages and contents, fix selection navigation etc.

### CODE REVIEW

**Comment by gflores-jahia** [comment]
Just trying to get context: for ContentListHeader, I see there's two versions: `ContentListHeader` and `ContentListHeaderMoon` and seems to be using former. I'm guessing we can't use moonstone version yet?

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.utils.js [hunk 1]
```diff
@@ -52,7 +52,20 @@ export const structureData = function (parentPath, dataForParentPath = []) {
     }
 };
 
-const adaptedRow = r => ({
+export const flattenTree = function (rows, selector = p => p) {
```
Comment (gflores-jahia) [inline src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.utils.js:55]: just as alternative implementation - I would actually remove the selector part and just use map instead when using it: `flattenTree(rows).map(p => p)`, for example. This decouples the transformation part, esp that it's optional in some cases.

**Review by gflores-jahia — CHANGES_REQUESTED** [review]

File: src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.jsx [hunk 2]
```diff
@@ -27,7 +25,7 @@ const styles = theme => ({
         minHeight: 0,
         width: '100%'
     },
-    content: {
+    CONTENT: {
```
[the remark this answers is not in the document]
Comment (AKarmanov) [reply src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.jsx:28]: Must have been autorename feature in intelij , it should be lowercase.

**Comment by AKarmanov** [comment]
> Just trying to get context: for ContentListHeader, I see there's two versions: `ContentListHeader` and `ContentListHeaderMoon` and seems to be using former. I'm guessing we can't use moonstone version yet?

We are using moonstone version already, the other one needs to be removed.

**Review by gflores-jahia — APPROVED** [review]
looks good!

**Pull request merged**